src/utils/hooks/useListApplications.tsx

```diff
@@ -1,98 +1,43 @@
 import { useEffect, useState } from 'react'
-import {
-  Application,
-  ApplicationStageStatusAll,
-  useGetApplicationsQuery,
-  useGetApplicationsStagesQuery,
-} from '../../utils/generated/graphql'
-import { ApplicationDetails } from '../types'
+import buildFilter from '../helpers/application/buildQueryFilters'
+import buildSortFields, { getPaginationVariables } from '../helpers/application/buildQueryVariables'
+import { useGetApplicationsListQuery, ApplicationList } from '../../utils/generated/graphql'
+import { QueryObject } from '../hooks/useRouter'
 
-interface UseListApplicationsProps {
-  type?: string
+export type URLQueryFilter = {
+  [key: string]: string
 }
 
-interface ApplicationDetailsMap {
-  [serial: string]: ApplicationDetails
-}
-
-const useListApplications = ({ type }: UseListApplicationsProps) => {
-  const [applications, setApplications] = useState<ApplicationDetailsMap>()
+const useListApplications = (urlFilters: QueryObject) => {
+  const [applications, setApplications] = useState<ApplicationList[]>([])
   const [loading, setLoading] = useState(true)
   const [error, setError] = useState('')
+  const { type, sortBy, page, perPage } = urlFilters
 
-  const { data, error: applicationsError } = useGetApplicationsQuery({
-    variables: { code: type as string },
+  const filters = buildFilter(urlFilters)
+  const sortFields = sortBy ? buildSortFields(sortBy) : []
+  const { paginationOffset, numberToFetch } = getPaginationVariables(
+    page ? Number(page) : 1,
+    perPage ? Number(perPage) : undefined
+  )
+
+  const { data, error: applicationsError } = useGetApplicationsListQuery({
+    variables: { filters, sortFields, paginationOffset, numberToFetch },
     fetchPolicy: 'network-only',
     skip: !type,
```

Comment:
Can this be removed?

Reply:
Remove what exactly? 🤷‍♂️

Reply:
I was talking about removing the `skip: !type`, but I'm not sure if it can be done

Reply:
Yeah I think we can remove this now. And that means I can implement your previous suggestion about destructuring in hook parameters, since this function no longer needs type specifically.

Good suggestion.

(Will quickly test before merging)